sympy/combinatorics/homomorphisms.py

```diff
@@ -417,3 +419,88 @@ def block_homomorphism(group, blocks):
     images = {g: Permutation([b[p[i]^g] for i in identity]) for g in group.generators}
     H = GroupHomomorphism(group, codomain, images)
     return H
+
+def group_isomorphism(G, H):
+    '''
+    Compute an isomorphism between 2 given groups.
+
+    Arguments:
+        G (a finite `FpGroup` or a `PermutationGroup`) -- First group
+        H (a finite `FpGroup` or a `PermutationGroup`) -- Second group
+
+    Returns:
+    If the groups are not isomorphic, `None` is returned.
+    If the groups are isomorphic, the corresponding `GroupHomomorphism` object is returned.
```

Comment:
"a corresponding `GroupHomomorphism` object" would be better, because isomorphisms are generally not unique.